### PR TITLE
Feature/chessboard

### LINKED ISSUE #121: Send and show moves
# Task
## Detailed Description

- The frontend communicats with the proxy to send moves.
- it displays moves that have been received via the proxy.
## Associated story

Story: #117 Play off-chain-chess without validation
## Notes

depends on #119 and #118
- [x] Factory function to sign and send moves
- [ ] Call function from game

### PR DESCRIPTION
**Pull request by ghost**
Closes #121

### CODE REVIEW

File: app/javascript/playGame.js [hunk 1]
```diff
@@ -220,25 +343,181 @@ angular.module('dappChess').controller('PlayGameCtrl',
...
+    // OFF-CHAIN -----------------------------------------------
+
+    function processChessMoveOffChain(chessMove) {
+
+      board.enableUserInput(!board.isUserInputEnabled());
+      console.log('Set Column color');
+      let game = $scope.getGame();
+      let highlights = lightItUp();
+
+      claimWin();
```
Comment (dennis-westphal) [inline app/javascript/playGame.js]: @wiilee: Is there a reason why you want to claim a win after every single move?

**Comment by dennis-westphal** [comment]
Needs to be checked after merge from dev first

File: app/javascript/playGame.js [hunk 2]
```diff
@@ -22,16 +49,134 @@ angular.module('dappChess').controller('PlayGameCtrl',
...
+          castling =  fenComponents[2],
+          enPassant =  fenComponents[3],
+          halfMoveClock = fenComponents[4],
+          fullMoveCounter = fenComponents[5];
+      console.log('board: ', board);
+      console.log('Color: ',activeColor);
+      console.log('enPAssant: ',enPassant);
+      console.log('castling: ',castling);
+      console.log('halfMove: ',halfMoveClock);
+      console.log('fullmove: ',fullMoveCounter);
```
Comment (kyroy) [inline app/javascript/playGame.js]: can we please remove all the unnecessary console.logs?

File: app/javascript/playGame.js [hunk 3]
```diff
@@ -22,16 +49,134 @@ angular.module('dappChess').controller('PlayGameCtrl',
...
+      // fill rest of shadow field
+      for (let j=0; j < 8; j++) {
+        state.push(new BigNumber(0));
+      }
+
+      // fullmove
+      console.log('high: ', Math.abs(parseInt(fullMoveCounter / 128)));
+      console.log('low: ',(parseInt(((fullMoveCounter % 128) - 1)/2)));
+      state[8] = new BigNumber(Math.abs(parseInt(fullMoveCounter / 128)));
+      state[9] = new BigNumber(parseInt(((fullMoveCounter % 128) - 1) / 2));
```
Comment (kyroy) [inline app/javascript/playGame.js]: moveCounter is not correctly merged. This was fixed by @HeavyPanda and me

File: static/index.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <!DOCTYPE html>
 <html data-ng-app="dappChess">
-  <head>
+  <head
```
Comment (kyroy) [inline static/index.html]: head tag is not closed (wrong merge)

File: app/javascript/playGame.js [hunk 5]
```diff
@@ -694,73 +600,48 @@ angular.module('dappChess').controller('PlayGameCtrl',
...
-          board = new Chessboard('my-board', {
-              position: currentFen,
-              eventHandlers: {
-                onPieceSelected: pieceSelected,
-                onMove: pieceMove
-              }
-            }
-          );
-
-          chess.load(currentFen);
```
Comment (kyroy) [inline app/javascript/playGame.js:731]: If you remove this line, chess has to be initiated with the fen.

File: app/javascript/playGame.js [hunk 6]
```diff
@@ -694,73 +600,48 @@ angular.module('dappChess').controller('PlayGameCtrl',
...
-          let gamer;
-          if (gameState[56].toNumber() === 1) {
-            gamer = 'white';
-          } else {
-            gamer = 'black';
+    let initGameOffChain = function(game) {
+      // init chess.js and chessboard.js
+      chess = new Chess();
+      board = new Chessboard('my-board', {
+          position: chess.fen(),
```
Comment (kyroy) [inline app/javascript/playGame.js:660]: before you can use the fen of chess, you have to set it from the current game

**Comment by graup** [comment]
Looks ok so far. Will merge and continue working on it with #184 / #201.

**Pull request merged**